apps/ledger-live-desktop/src/renderer/families/celo/ActivateFlowModal/Body.js

```diff
@@ -0,0 +1,202 @@
+import { compose } from "redux";
+import { connect, useDispatch } from "react-redux";
+import { Trans, withTranslation } from "react-i18next";
+import { createStructuredSelector } from "reselect";
+
+import { UserRefusedOnDevice } from "@ledgerhq/errors";
+import { addPendingOperation } from "@ledgerhq/live-common/lib/account";
+import { getAccountBridge } from "@ledgerhq/live-common/lib/bridge";
+import { SyncSkipUnderPriority } from "@ledgerhq/live-common/lib/bridge/react";
+import useBridgeTransaction from "@ledgerhq/live-common/lib/bridge/useBridgeTransaction";
+
+import type { Account, Operation } from "@ledgerhq/live-common/lib/types";
+import type { TFunction } from "react-i18next";
+import type { Device } from "@ledgerhq/live-common/lib/hw/actions/types";
+import type { StepId, StepProps, St } from "./types";
+
+import { updateAccountWithUpdater } from "~/renderer/actions/accounts";
+
+import Track from "~/renderer/analytics/Track";
+import { getCurrentDevice } from "~/renderer/reducers/devices";
+import { closeModal, openModal } from "~/renderer/actions/modals";
+
+import Stepper from "~/renderer/components/Stepper";
+import StepVote, { StepVoteFooter } from "./steps/StepVote";
+import GenericStepConnectDevice from "~/renderer/modals/Send/steps/GenericStepConnectDevice";
+import StepConfirmation, { StepConfirmationFooter } from "./steps/StepConfirmation";
+import logger from "~/logger/logger";
+import { CeloVote } from "@ledgerhq/live-common/lib/families/celo/types";
+
+type OwnProps = {|
+  stepId: StepId,
+  onClose: () => void,
+  onChangeStepId: StepId => void,
+  params: {
+    account: Account,
+    parentAccount: ?Account,
+    vote: CeloVote,
+  },
+  name: string,
+|};
+
+type StateProps = {|
+  t: TFunction,
+  device: ?Device,
+  accounts: Account[],
+  device: ?Device,
+  closeModal: string => void,
+  openModal: string => void,
+|};
+
+type Props = OwnProps & StateProps;
+
+const steps: Array<St> = [
+  {
+    id: "vote",
+    label: <Trans i18nKey="celo.activate.steps.vote.title" />,
+    component: StepVote,
+    noScroll: true,
+    footer: StepVoteFooter,
+  },
+  {
+    id: "connectDevice",
+    label: <Trans i18nKey="celo.activate.steps.connectDevice.title" />,
+    component: GenericStepConnectDevice,
+    onBack: ({ transitionTo }: StepProps) => transitionTo("vote"),
+  },
+  {
+    id: "confirmation",
+    label: <Trans i18nKey="celo.activate.steps.confirmation.title" />,
+    component: StepConfirmation,
+    footer: StepConfirmationFooter,
+  },
+];
+
+const mapStateToProps = createStructuredSelector({
+  device: getCurrentDevice,
+});
+
+const mapDispatchToProps = {
+  closeModal,
+  openModal,
+};
+
+const Body = ({
+  t,
+  stepId,
+  device,
+  closeModal,
+  openModal,
+  onChangeStepId,
+  params,
+  name,
+}: Props) => {
+  const [optimisticOperation, setOptimisticOperation] = useState(null);
+  const [transactionError, setTransactionError] = useState(null);
```

Comment:
Minor: we could type the `useStates` above for a little bit of extra type safety.  Something like:

```
const [optimisticOperation, setOptimisticOperation] = useState<Operation | null>(null);
  const [transactionError, setTransactionError] = useState<Error | null>(null);
```